hki worksheet.rb

```diff
@@ -0,0 +1,121 @@
+
+
+# Step 1: Establish the layers
+
+# In this section of the file, as a series of comments,
+# create a list of the layers you identify.
+# Which layers are nested in each other?
+# Which layers of data "have" within it a different layer?
+# Which layers are "next" to each other?
+
+# top level = Array of drivers
+# 2nd level = hash for each driver w/ unique trips (date + rider ID), amounts earned (cost of trips), rating
+# 3rd level = array within each hash for each driver - arrays: unique trips, amounts earned, rating
+
+########################################################
+# Step 2: Assign a data structure to each layer
+
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+
+########################################################
+
+# Step 3: Make the data structure!
+
+driver_data = [
+  {
+    driver_id: "DR0001",
+    unique_trips: [("3rd Feb 2016" + "RD0003"), ("3rd Feb 2016" + "RD0015"), ("5th Feb 2016" + "RD0003")],  #tried to use this a proxy for a unique trip
```

Comment:
This is an clever solution for identifying unique trips.

Can you think of any downsides to doing this if you had to solve additional problems with this data?